Detectors/GlobalTracking/src/MatchTOF.cxx

```diff
@@ -207,6 +235,86 @@ int MatchTOF::prepareInteractionTimes()
   // do nothing. If you think it can be useful have a look at MatchTPCITS
   return 0;
 }
+//______________________________________________
+void MatchTOF::printGrouping(const std::vector<o2::dataformats::MatchInfoTOFReco>& origin, const std::vector<std::vector<o2::dataformats::MatchInfoTOFReco>>& grouped)
+{
+  printf("Original vector\n");
+  for (const auto& seed : origin) {
+    printf("Pair: track=%d TOFcl=%d\n", seed.getIdLocal(), seed.getTOFClIndex());
+  }
+
+  printf("\nGroups\n");
+  int ngroups = 0;
+  for (const auto& gr : grouped) {
+    ngroups++;
+    printf("Group %d\n", ngroups);
+    for (const auto& seed : gr) {
+      printf("Pair: track=%d TOFcl=%d\n", seed.getIdLocal(), seed.getTOFClIndex());
+    }
+  }
+}
+//______________________________________________
+void MatchTOF::groupingMatch(std::vector<o2::dataformats::MatchInfoTOFReco> origin, std::vector<std::vector<o2::dataformats::MatchInfoTOFReco>>& grouped, std::vector<std::vector<int>>& firstEls, std::vector<std::vector<int>>& secondEls)
```

Comment:
You are passing `origin` by copy, i.e. the results of its modification are lost. Is this what you want?
This may create some overheads.

Reply:
Yes, since I need to modify it (e.g. erasing object) without affecting the original vector at least for the moment which is not in production.
In future I could release this condition if I will use the output of the grouping instead of the original one for the next steps.

Reply:
I will pass origin as a reference since I changed accordingly to the next comment. Thanks